### ISSUE TITLE
Running list of errors for each data set up to .friedman.2014

### ISSUE THREAD

**Issue by mrhelmus** [willpearse/natdb#73]
Here is a running list of problems with NATDB downloading on a PC. I give a description of what I think is going on, and then below that the code and the error.

First, there are lots that work. Basically anything that is a csv or a txt works. I stopped adding the ones that work.
```
# work fine -------------------------------------------
# .albouy.2015
data <- read.csv(unzip(suppdata("E096-203","Functional_data.zip", "esa_archives"), "Functional_data.csv"), sep=";")

# .anderson.2015
data<-read.csv(suppdata("10.1371/journal.pone.0166714",2))

# .artacho.2015
data <- read.csv2(suppdata("10.5061/dryad.qg062", "phenotypictraits.csv"), sep=';')

# .augspurger.2016a
  data <- read.csv(suppdata("10.5061/dryad.56cn4","Data File 1. Diaspore traits.csv"), as.is=TRUE)

# .bengtsson.2016
  data <- read.csv(suppdata('10.5061/dryad.62054', 'bengtsson_etal_2016_traits.csv'), as.is=TRUE)

# .comeault.2013
  data <- read.table(suppdata("10.5061/dryad.ck2cm","Tcris_FHA_phenotypes.txt"),header=TRUE)

# .edwards.2015a
    data <- read.csv(suppdata("E096-202", "Table1.csv", "esa_archives"))
# stopped adding ones that worked    
```

ERROR 1

This is probabally the hardest error. It is with read_excel and suppdata with dryad download. I tried to just use suppdata to download xls from dryad and could not open the downloaded files... so there is some interaction going on here.

```
# does not work -----------------------------------------------------------------

# .ameztegui.2016
data <- as.data.frame(read_excel(suppdata("10.5061/dryad.12b0h/2","FunctionalTraits_Dryad.xlsx")))

# .arnold.2016
data <- as.data.frame(read_excel(suppdata("10.5061/dryad.t3d52", "Arnold_etal_2016_functecol_dataset.xlsx"), as.is = TRUE, skip = 3))

# .aubret.2012a
data <- as.data.frame(read_excel(suppdata("10.5061/dryad.14cr5345", "Aubret%2053172.xlsx"),  sheet=1))


#all of the above give this error

Error in read_fun(path = path, sheet = sheet, limits = limits, shim = shim,  :
  Evaluation error: error -103 with zipfile in unzGetCurrentFileInfo
.
```
ERROR 2

This one just runs and runs. It is like the above one but it deals with figshare. So I think the problem is the suppdata and read_excel interaction (it does not matter where you are downloading from). If you can fix this interaction then ERRORS 1 and 2 might go away...

```
# .bello_bedoy.2015b
data <- as.data.frame(read_excel(
        suppdata("10.6084/m9.figshare.1190766.v2","mating occurrencePzoe_2010.xls")
    ))    
```
ERROR 3
Problem with the suppdata function not the unzip I think. It is having a problem with wiley... is it because these are not open access?
```
# .benesh.2017
data <- read.csv(unzip(suppdata("10.1002/ecy.1680", 1), "CLC_database_lifehistory.csv"))
Error in download.file(url, destination, quiet = TRUE) : 
  cannot open URL 'https://onlinelibrary.wiley.com/action/'
In addition: Warning message:
In download.file(url, destination, quiet = TRUE) :
  cannot open URL 'https://onlinelibrary.wiley.com/action/': HTTP status was '404 Not Found'

# .engemann.2016
 data <- read.delim(unzip(suppdata("10.1002/ecy.1569", 1), "DataS1/GrowthForm_Final.txt"))
 Error in download.file(url, destination, quiet = TRUE) : 
  cannot open URL 'https://onlinelibrary.wiley.com/action/'
In addition: Warning message:
In download.file(url, destination, quiet = TRUE) :
  cannot open URL 'https://onlinelibrary.wiley.com/action/': HTTP status was '404 Not Found'
```

ERROR 4
this one needs a file extension

```
# .cariveau.2016
data <- as.data.frame(read_excel(suppdata("10.1371/journal.pone.0151482", 3), sheet="TableS1_v2"))
Error: Missing file extension.
```

ERROR 5
This one crashes R and gives me a bomb in R studio!

```
.friedman.2014 <- function(...){
    data <- as.data.frame(read_excel(suppdata("10.5061/dryad.489c7","NILs_rawdata.xls")))

 fread: wanted 1 got 0 loc=570826752
 seek: wanted to seek to sector...
THEN R ABORTS

```

OK I got through to .friedman.2014 ...

**Comment by willpearse** [comment]
Wow. I think a lot of these (all of them?) are being caused by
`read_excel`. This serves me right for just find-replacing everything to
swap it over.

I don't have time to do this today or tomorrow, but Friday I will take a
look through. That's assuming this isn't going to cause a problem in class
for you Thursday Matt, right?

**Comment by mrhelmus** [comment]
Yes, it is no problem at all if you do not get to this anytime soon. 

I am going to have the class code for csv/txt files, not xls or zips to make things easier. I had them all fork your package... so now NATDB is probably one of your most forked!

**Comment by willpearse** [comment]
Oh my, 34 forks...

Makes sense to get them to focus on plain text files. It's really
frustrating that `gdata` seems so difficult for Windows users to install,
because it's been working perfectly for me as an Excel loader into R for as
long as I can remember :-(

**Comment by willpearse** [comment]
This has been very helpful. I've now fixed, I hope, the last of the `readxl` bugs.

What remains is a more profound problem: Wiley have made the new ESA Data Archives papers even harder to download. This is a problem for `suppdata`, and I'm working on this from that end. Until then, I've commented-out the functions that are currently broken, and I'll add them back in once they're not...

Once this builds on Travis (the badge on the repo should show that) then everything should, in principle, work. Or, at least, it does on my machine...
